### ISSUE TITLE
Allow "Group achievements with data from Steam Hunters" as a toggle on the page itself

### ISSUE THREAD

**Issue by VenatusSimpleX** [SteamDatabase/BrowserExtension#188]
### Is your feature request related to a problem? Please describe.

Some games have a lot achievements that require DLCs, while some games don't. The grouping feature by default groups the achievements by DLC and Update, which makes me very confused, because sometimes, I want them grouped so I know what achievements I can go for (ones that don't need me to buy the DLC) or ungrouped so I don't get confused by Updates.

One example is Saints Row https://steamcommunity.com/stats/742420/achievements , with this feature turned on, it's awesome because now I know what achievements to avoid, since I only own the base game.

But for games like... https://steamcommunity.com/stats/733460/achievements , now I get very confused because the achievements are grouped by updates, so I'd prefer to have the feature turned off at that moment.

Currently, based on the game I am now playing, I turn the feature on or off manually through the extension's options, which makes it a bit of a hassle.

### Describe the solution you'd like

Add a button next to the "Collapse all groups" button that allows the page to be groups or ungrouped, so it'll be just a click away from grouping or ungrouping the achievements.

**Comment by miguel-is-coding** [comment]
Hi @xPaw I was thinking about helping with my contribution on this. Would this be a good addition?

Thanks.

**Comment by xPaw** [comment]
Probably fine to add, the only problem is you won't know if achievements can be grouped for a specific game while the setting is disabled. So enabling it from the page may not necessarily produce any groups.

**Comment by miguel-is-coding** [comment]
Would be ideal if this proposed button only appears when the "Group with Steam Hunters data" setting is enabled?

**Comment by xPaw** [comment]
For that to work, disabling from the page would have to be separate to the setting.

**Comment by miguel-is-coding** [comment]
I get it, thought it was related. I will check the code first to have a better view about what could be done 👍🏼

**Comment by miguel-is-coding** [comment]
Hey @xPaw, I've been thinking about this issue these days, and I wanted to share what could be a better approach for this.

Instead of having a button to group or ungroup achievements (which only may apply to this issue), I realized that many games which are only updated through time adding new content, follow the same pattern when adding achievements, which is creating:

`Update #<UpdateNumber>`

And so on.

So I think an idea will be, when having the `achievementUpdates` object after fetching them, map all the achievementUpdate which are related to updates (and are always tagged as `Update #`), and add them to the base game.

What do you think?

**Comment by xPaw** [comment]
Doesn't that defeat the purpose?

**Comment by miguel-is-coding** [comment]
Not sure, because I was thinking more about the problem @VenatusSimpleX exposed instead of the solution proposed: to not get confused with the updates and not mix them with the DLC's.

Maybe you have in mind a better approach? I'll be glad to help 💪

**Comment by xPaw** [comment]
Well grouping updates is kind of part of the feature, partly because grouping by dlc requires manual tagging by steamhunters.

**Comment by miguel-is-coding** [comment]
Yeah, makes sense. I think that I've mixed two user groups in this case, users as Venatus which would like to be able to see the base game and the updates altogether, on the other group which would prefer to have the updates separated from the base game, as that's the feature you mentioned from steamhunters.

This is the usual case of "This is not a bug, is a feature" 😄 

In that case, and back to the original answer you mentioned:

> Probably fine to add, the only problem is you won't know if achievements can be grouped for a specific game while the setting is disabled. So enabling it from the page may not necessarily produce any groups.

It looks like it would be needed to decouple the action to group/ungroup achievements from the setting, so if the option is disabled, we should add the action to fetch the data from Steamhunters and then group the achievements if needed.

Maybe the major issue here is the refactor that may be needed to be able to add this action and the impact it may have.

Am I right?

**Comment by xPaw** [comment]
> so if the option is disabled, we should add the action to fetch the data from Steamhunters

The issue here is that if the option is disabled, I would expect it to not send any requests related to the disabled feature.

**Comment by miguel-is-coding** [comment]
You are right, good point.

Then I think in summary  there are two different flows:
- **Steam Hunters achievements grouping is disabled in the extension options**.
  - This means that when I check a game's achievements, this will look like Steam UI (no grouping changes).
- **Steam Hunters achievements grouping is enabled**
  - Button to switch grouping is enabled/showed:
    - If the user wants to see all the achievements separated as groups per DLC/Updates -> They keep the switch button enabled.
    - If the user doesn't want to separate the group and have all the achievements in a single list like the original Steam UI -> They toggle the switch button to be disabled.

This switch button although should not affect the behaviour from the settings, as it only should affect the page from the game's achievements.
